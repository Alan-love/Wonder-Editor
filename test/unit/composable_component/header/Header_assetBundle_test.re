open Wonder_jest;

open Expect;

open Expect.Operators;

open Sinon;

open Js.Promise;

let _ =
  describe("Header AssetBundle", () => {
    let sandbox = getSandboxDefaultVal();

    beforeEach(() => {
      sandbox := createSandbox();

      MainEditorSceneTool.initState(~sandbox, ());

      MainEditorSceneTool.initInspectorEngineState(
        ~sandbox,
        ~isInitJob=false,
        ~noWorkerJobRecord=
          NoWorkerJobConfigToolEngine.buildNoWorkerJobConfig(
            ~initPipelines=
              {|
             [
              {
                "name": "default",
                "jobs": [
                    {"name": "init_inspector_engine" }
                ]
              }
            ]
             |},
            ~initJobs=
              {|
             [
                {"name": "init_inspector_engine" }
             ]
             |},
            (),
          ),
        (),
      );

      StateInspectorEngineService.unsafeGetState()
      |> MainUtils._handleInspectorEngineState
      |> StateInspectorEngineService.setState
      |> ignore;

      CanvasTool.prepareInspectorCanvasAndImgCanvas(sandbox) |> ignore;

      MainEditorSceneTool.prepareScene(sandbox);
    });
    afterEach(() => restoreSandbox(refJsObjToSandbox(sandbox^)));

    describe("test generate single rab", () =>
      describe("test buildSelectTreeForGenerateSingleRAB", () => {
        /* TODO add more tests */

        let truckWDBArrayBuffer = ref(Obj.magic(1));

        beforeAll(() =>
          truckWDBArrayBuffer := WDBTool.convertGLBToWDB("CesiumMilkTruck")
        );

        beforeEach(() => {
          MainEditorAssetTool.buildFakeFileReader();
          MainEditorAssetTool.buildFakeImage();

          LoadTool.buildFakeTextDecoder(LoadTool.convertUint8ArrayToBuffer);
          LoadTool.buildFakeURL(sandbox^);

          LoadTool.buildFakeLoadImage(.);
        });

        testPromise("test1", () => {
          let addedMaterialNodeId1 = MainEditorAssetIdTool.getNewAssetId();
          MainEditorAssetHeaderOperateNodeTool.addMaterial();

          let addedFolderNodeId1 = MainEditorAssetIdTool.getNewAssetId();
          MainEditorAssetHeaderOperateNodeTool.addFolder();

          MainEditorAssetTreeTool.Select.selectFolderNode(
            ~nodeId=addedFolderNodeId1,
            (),
          );

          MainEditorAssetUploadTool.loadOneTexture(
            ~imgName="image1.png",
            ~imgSrc="newImgBase64",
            (),
          )
          |> then_(uploadedTextureNodeId =>
               MainEditorAssetUploadTool.loadOneWDB(
                 ~fileName="Truck",
                 ~arrayBuffer=truckWDBArrayBuffer^,
                 (),
               )
               |> then_(uploadedWDBNodeId =>
                    HeaderAssetBundleTool.buildGenerateSingleRABModal(
                      ~selectTree=
                        HeaderAssetBundleTool.buildSelectTreeForGenerateSingleRAB
                        |> StateLogicService.getStateToGetData,
                      ~send=SinonTool.createOneLengthStub(sandbox^),
                      (),
                    )
                    |> BuildComponentTool.buildUI
                    |> ReactTestTool.createSnapshotAndMatch
                    |> resolve
                  )
             );
        });
        test("test2", () => {
          MainEditorAssetHeaderOperateNodeTool.addFolder();

          MainEditorAssetHeaderOperateNodeTool.addScriptEventFunction();

          HeaderAssetBundleTool.buildGenerateSingleRABModal(
            ~selectTree=
              HeaderAssetBundleTool.buildSelectTreeForGenerateSingleRAB
              |> StateLogicService.getStateToGetData,
            ~send=SinonTool.createOneLengthStub(sandbox^),
            (),
          )
          |> BuildComponentTool.buildUI
          |> ReactTestTool.createSnapshotAndMatch
        });
      })
    );
  });