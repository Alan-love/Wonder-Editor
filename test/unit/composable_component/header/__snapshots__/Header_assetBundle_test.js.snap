// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header AssetBundle test generate single rab test buildSelectTreeForGenerateSingleRAB test1 1`] = `
<article
  className="wonder-modal"
>
  <div
    className="modal-item"
  >
    <div
      className="modal-item-header"
    >
      Generate Single RAB
      <img
        onClick={[Function]}
        src="./public/img/close.png"
      />
    </div>
    <div
      className="modal-item-content"
    >
      <article
        className="wonder-selectTree"
      >
        <ul
          className="wonder-tree-node"
        >
          <li>
            <div
              className="item-triangle"
            >
              <img
                src="./public/img/down.png"
              />
            </div>
            <div
              className="select-box"
            >
              <input
                defaultChecked={false}
                onClick={[Function]}
                type="checkbox"
              />
            </div>
            <img
              className="treeNode-icon"
              src="./public/img/package.png"
            />
            <span>
              Assets
            </span>
          </li>
          <ul
            className="wonder-tree-node"
          >
            <li>
              <div
                className="item-triangle"
              />
              <div
                className="select-box"
              >
                <input
                  defaultChecked={false}
                  onClick={[Function]}
                  type="checkbox"
                />
              </div>
              <img
                className="treeNode-icon"
                src="./public/img/mat.jpg"
              />
              <span>
                New Material
              </span>
            </li>
          </ul>
          <ul
            className="wonder-tree-node"
          >
            <li>
              <div
                className="item-triangle"
              >
                <img
                  src="./public/img/down.png"
                />
              </div>
              <div
                className="select-box"
              >
                <input
                  defaultChecked={false}
                  onClick={[Function]}
                  type="checkbox"
                />
              </div>
              <img
                className="treeNode-icon"
                src="./public/img/package.png"
              />
              <span>
                New Folder
              </span>
            </li>
            <ul
              className="wonder-tree-node"
            >
              <li>
                <div
                  className="item-triangle"
                />
                <div
                  className="select-box"
                >
                  <input
                    defaultChecked={false}
                    onClick={[Function]}
                    type="checkbox"
                  />
                </div>
                <img
                  className="treeNode-icon"
                  src="newImgBase64"
                />
                <span>
                  image1
                </span>
              </li>
            </ul>
            <ul
              className="wonder-tree-node"
            >
              <li>
                <div
                  className="item-triangle"
                >
                  <img
                    src="./public/img/down.png"
                  />
                </div>
                <div
                  className="select-box"
                >
                  <input
                    defaultChecked={false}
                    onClick={[Function]}
                    type="checkbox"
                  />
                </div>
                <img
                  className="treeNode-icon"
                  src="./public/img/package.png"
                />
                <span>
                  Truck_Geometrys
                </span>
              </li>
              <ul
                className="wonder-tree-node"
              >
                <li>
                  <div
                    className="item-triangle"
                  />
                  <div
                    className="select-box"
                  >
                    <input
                      defaultChecked={false}
                      onClick={[Function]}
                      type="checkbox"
                    />
                  </div>
                  <img
                    className="treeNode-icon"
                    src="./public/img/wdb.png"
                  />
                  <span>
                    Cesium_Milk_Truck_0
                  </span>
                </li>
              </ul>
              <ul
                className="wonder-tree-node"
              >
                <li>
                  <div
                    className="item-triangle"
                  />
                  <div
                    className="select-box"
                  >
                    <input
                      defaultChecked={false}
                      onClick={[Function]}
                      type="checkbox"
                    />
                  </div>
                  <img
                    className="treeNode-icon"
                    src="./public/img/wdb.png"
                  />
                  <span>
                    Cesium_Milk_Truck_1
                  </span>
                </li>
              </ul>
              <ul
                className="wonder-tree-node"
              >
                <li>
                  <div
                    className="item-triangle"
                  />
                  <div
                    className="select-box"
                  >
                    <input
                      defaultChecked={false}
                      onClick={[Function]}
                      type="checkbox"
                    />
                  </div>
                  <img
                    className="treeNode-icon"
                    src="./public/img/wdb.png"
                  />
                  <span>
                    Cesium_Milk_Truck_2
                  </span>
                </li>
              </ul>
              <ul
                className="wonder-tree-node"
              >
                <li>
                  <div
                    className="item-triangle"
                  />
                  <div
                    className="select-box"
                  >
                    <input
                      defaultChecked={false}
                      onClick={[Function]}
                      type="checkbox"
                    />
                  </div>
                  <img
                    className="treeNode-icon"
                    src="./public/img/wdb.png"
                  />
                  <span>
                    Wheels
                  </span>
                </li>
              </ul>
              <ul
                className="wonder-tree-node"
              >
                <li>
                  <div
                    className="item-triangle"
                  />
                  <div
                    className="select-box"
                  >
                    <input
                      defaultChecked={false}
                      onClick={[Function]}
                      type="checkbox"
                    />
                  </div>
                  <img
                    className="treeNode-icon"
                    src="./public/img/wdb.png"
                  />
                  <span>
                    Wheels
                  </span>
                </li>
              </ul>
            </ul>
            <ul
              className="wonder-tree-node"
            >
              <li>
                <div
                  className="item-triangle"
                >
                  <img
                    src="./public/img/down.png"
                  />
                </div>
                <div
                  className="select-box"
                >
                  <input
                    defaultChecked={false}
                    onClick={[Function]}
                    type="checkbox"
                  />
                </div>
                <img
                  className="treeNode-icon"
                  src="./public/img/package.png"
                />
                <span>
                  Materials
                </span>
              </li>
              <ul
                className="wonder-tree-node"
              >
                <li>
                  <div
                    className="item-triangle"
                  />
                  <div
                    className="select-box"
                  >
                    <input
                      defaultChecked={false}
                      onClick={[Function]}
                      type="checkbox"
                    />
                  </div>
                  <img
                    className="treeNode-icon"
                    src="data:image/png;base64 for img-canvas"
                  />
                  <span>
                    truck
                  </span>
                </li>
              </ul>
              <ul
                className="wonder-tree-node"
              >
                <li>
                  <div
                    className="item-triangle"
                  />
                  <div
                    className="select-box"
                  >
                    <input
                      defaultChecked={false}
                      onClick={[Function]}
                      type="checkbox"
                    />
                  </div>
                  <img
                    className="treeNode-icon"
                    src="data:image/png;base64 for img-canvas"
                  />
                  <span>
                    glass
                  </span>
                </li>
              </ul>
              <ul
                className="wonder-tree-node"
              >
                <li>
                  <div
                    className="item-triangle"
                  />
                  <div
                    className="select-box"
                  >
                    <input
                      defaultChecked={false}
                      onClick={[Function]}
                      type="checkbox"
                    />
                  </div>
                  <img
                    className="treeNode-icon"
                    src="data:image/png;base64 for img-canvas"
                  />
                  <span>
                    window_trim
                  </span>
                </li>
              </ul>
              <ul
                className="wonder-tree-node"
              >
                <li>
                  <div
                    className="item-triangle"
                  />
                  <div
                    className="select-box"
                  >
                    <input
                      defaultChecked={false}
                      onClick={[Function]}
                      type="checkbox"
                    />
                  </div>
                  <img
                    className="treeNode-icon"
                    src="data:image/png;base64 for img-canvas"
                  />
                  <span>
                    wheels
                  </span>
                </li>
              </ul>
            </ul>
            <ul
              className="wonder-tree-node"
            >
              <li>
                <div
                  className="item-triangle"
                >
                  <img
                    src="./public/img/down.png"
                  />
                </div>
                <div
                  className="select-box"
                >
                  <input
                    defaultChecked={false}
                    onClick={[Function]}
                    type="checkbox"
                  />
                </div>
                <img
                  className="treeNode-icon"
                  src="./public/img/package.png"
                />
                <span>
                  Textures
                </span>
              </li>
              <ul
                className="wonder-tree-node"
              >
                <li>
                  <div
                    className="item-triangle"
                  />
                  <div
                    className="select-box"
                  >
                    <input
                      defaultChecked={false}
                      onClick={[Function]}
                      type="checkbox"
                    />
                  </div>
                  <img
                    className="treeNode-icon"
                    src="./public/null.png"
                  />
                  <span>
                    texture_0
                  </span>
                </li>
              </ul>
              <ul
                className="wonder-tree-node"
              >
                <li>
                  <div
                    className="item-triangle"
                  />
                  <div
                    className="select-box"
                  >
                    <input
                      defaultChecked={false}
                      onClick={[Function]}
                      type="checkbox"
                    />
                  </div>
                  <img
                    className="treeNode-icon"
                    src="./public/null.png"
                  />
                  <span>
                    texture_1
                  </span>
                </li>
              </ul>
            </ul>
          </ul>
        </ul>
      </article>
    </div>
    <div
      className="modal-item-footer"
    >
      <button
        className="footer-submit"
        onClick={[Function]}
      >
        OK
      </button>
    </div>
  </div>
</article>
`;

exports[`Header AssetBundle test generate single rab test buildSelectTreeForGenerateSingleRAB test2 1`] = `
<article
  className="wonder-modal"
>
  <div
    className="modal-item"
  >
    <div
      className="modal-item-header"
    >
      Generate Single RAB
      <img
        onClick={[Function]}
        src="./public/img/close.png"
      />
    </div>
    <div
      className="modal-item-content"
    >
      <article
        className="wonder-selectTree"
      >
        <ul
          className="wonder-tree-node"
        >
          <li>
            <div
              className="item-triangle"
            >
              <img
                src="./public/img/down.png"
              />
            </div>
            <div
              className="select-box"
            >
              <input
                defaultChecked={false}
                onClick={[Function]}
                type="checkbox"
              />
            </div>
            <img
              className="treeNode-icon"
              src="./public/img/package.png"
            />
            <span>
              Assets
            </span>
          </li>
          <ul
            className="wonder-tree-node"
          >
            <li>
              <div
                className="item-triangle"
              />
              <div
                className="select-box"
              >
                <input
                  defaultChecked={false}
                  onClick={[Function]}
                  type="checkbox"
                />
              </div>
              <img
                className="treeNode-icon"
                src="./public/img/package.png"
              />
              <span>
                New Folder
              </span>
            </li>
          </ul>
          <ul
            className="wonder-tree-node"
          >
            <li>
              <div
                className="item-triangle"
              />
              <div
                className="select-box"
              >
                <input
                  defaultChecked={false}
                  onClick={[Function]}
                  type="checkbox"
                />
              </div>
              <img
                className="treeNode-icon"
                src="./public/img/scriptEventFunction.png"
              />
              <span>
                New Script EventFunction
              </span>
            </li>
          </ul>
        </ul>
      </article>
    </div>
    <div
      className="modal-item-footer"
    >
      <button
        className="footer-submit"
        onClick={[Function]}
      >
        OK
      </button>
    </div>
  </div>
</article>
`;
